src/osd.c

```diff
@@ -364,6 +378,31 @@ render_osd(struct server *server, cairo_t *cairo, int w, int h,
 			case LAB_FIELD_TYPE:
 				buf_add(&buf, get_type(*view));
 				break;
+			case LAB_FIELD_TYPE_SHORT:
+				buf_add(&buf, get_type_short(*view));
+				break;
+			case LAB_FIELD_WORKSPACE:
+				buf_add(&buf, (*view)->workspace->name);
+				break;
+			case LAB_FIELD_WIN_STATE:
+				if ((*view)->maximized) {
+					buf_add(&buf, "M");
+				} else if ((*view)->minimized) {
+					buf_add(&buf, "m");
+				} else if ((*view)->fullscreen) {
+					buf_add(&buf, "F");
+				} else {
+					buf_add(&buf, " ");
```

Comment:
Do we need this alternate case here? Same question for the `LAB_FIELD_OUTPUT` case below.
Other than that, this LGTM.

Reply:
I see, without it we may reuse the earlier buffer results because we only reset `buf.len`. We need a `buf_clear(buf)` that also resets the first byte of the internal allocation to `\0`.
Feel free to leave this workaround in for now, we can add the `buf_clear()` in a follow up PR.

Reply:
Yeah, I noticed when I had output set, but nothing was produced due to the if, it repeated what was in the buffer, but I hadn't investigated. The one for  state, I kept so that the output would
stay aligned (I use a mono font)